src/api/app/models/project.rb

```diff
@@ -1404,13 +1404,10 @@ def bsrequest_repos_map(project)
 
   def self.valid_name?(name)
```

Comment:
The problem of using `format_of` is that we should write a regular expression that match, and here for some conditions I think it is match clear to write a regular expression that doesn't match, like we do with `return false if name =~ /::/`

And it is not only used as validation, for example:

- https://github.com/openSUSE/open-build-service/blob/2889f4d98df379f29bfe3cc47ec9eadf0eff259b/src/api/app/helpers/validation_helper.rb#L11

- https://github.com/openSUSE/open-build-service/blob/master/src/api/app/models/branch_package.rb#L589

- https://github.com/openSUSE/open-build-service/blob/3271934e216593ddaf2f3576a125abdb3d707d6c/src/api/app/jobs/consistency_check.rb#L18